source/common/modules/markdown-editor/plugins/render-codeblock-button.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        Emphasis rendering plugin
+ * CVM-Role:        CodeMirror Plugin
+ * Maintainer:      Hendrik Erz
+ * License:         GNU GPL v3
+ *
+ * Description:     This plugin renders copy-to-clipboard button
+ *
+ * END HEADER
+ */
+
+import CodeMirror, { commands } from 'codemirror'
+
+const clipboard = window.clipboard
+/**
+   * Declare the markdownRenderEmphasis command
+   *
+   * @param   {CodeMirror.Editor}  cm  The CodeMirror instance
+*/
+;(commands as any).markdownRenderCodeBlockButton = function (cm: CodeMirror.Editor) {
+  cm.startOperation()
+  let countCodeblock = 0
+  const codeBlockRE = /^(?:\s{0,3}`{3}|~{3}).*/
+  const lineCount = cm.lineCount()
+  let incodeblock = false
+  let codesblocks = new Array<string>()
+  let codeblock = ''
+  const CodeblockOpenCount = document.getElementsByClassName('code-block-first-line').length
+  const CodeblockCloseCount = document.getElementsByClassName('code-block-last-line').length
+  for (let j = 0; j < lineCount; j++) {
+    const line = cm.getLine(j)
+    if (codeBlockRE.test(line) && !incodeblock) {
+      countCodeblock = countCodeblock + 1
+      codeblock = codeblock + cm.getLine(++j) + '\n'
+      incodeblock = true
+    } else if (codeBlockRE.test(line) && incodeblock) {
+      codesblocks.push(codeblock)
```

Comment:
I noticed that the copied code always had an additional newline at the end, which we should remove before pushing the codeblock onto the array:
```suggestion
      // Remove newline character after the last code line
      codeblock = codeblock.slice(0, -1)
      codesblocks.push(codeblock)
```